openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -81,50 +82,53 @@
  *
  * @since 1.9.0
  */
-public class OpenAPIBuildToolRunner implements BuildToolRunner {
-    static List<Diagnostic> diagnostics = new ArrayList<>();
+public class OpenAPICodeGeneratorTool implements CodeGeneratorTool {
+    static String md5HexOpenAPI;
 
     @Override
-    public String getToolName() {
+    public String toolName() {
         return "openapi";
     }
 
     @Override
-    public List<Diagnostic> diagnostics() {
-        return diagnostics;
-    }
-
-    @Override
-    public void executeTool(ToolContext toolContext) {
-
+    public void execute(ToolContext toolContext) {
         ImmutablePair<OASClientConfig, OASServiceMetadata> codeGeneratorConfig;
-        TomlNodeLocation location = toolContext.packageInstance().ballerinaToml().get().tomlAstNode().location();
+        TomlNodeLocation location = toolContext.currentPackage().ballerinaToml().get().tomlAstNode().location();
         try {
             // Validate the OAS file whether we can handle within OpenAPI tool
             if (!canHandle(toolContext)) {
                 Constants.DiagnosticMessages error = Constants.DiagnosticMessages.WARNING_FOR_UNSUPPORTED_CONTRACT;
-                reportDiagnostics(error, error.getDescription(), location);
+                createDiagnostics(toolContext, error, error.getDescription(), location);
                 return;
             }
+            // Handle the code generation
             String oasFilePath = toolContext.filePath();
-            Path packagePath = toolContext.packageInstance().project().sourceRoot();
-            OpenAPI openAPI = getOpenAPIContract(packagePath, Path.of(oasFilePath), location);
-            if (openAPI == null) {
+            Path packagePath = toolContext.currentPackage().project().sourceRoot();
+            Optional<OpenAPI> openAPI = getOpenAPIContract(packagePath, Path.of(oasFilePath), location, toolContext);
+            if (openAPI.isEmpty()) {
                 return;
             }
+            // Handle the details with Ballerina.toml option table, if option table is null then by default
+            // client will generate
```

Comment:
can we improve this comment? not sure what happens here

Reply:
Please refer : https://github.com/ballerina-platform/openapi-tools/pull/1598/files